### PR TITLE
refactor #14505 to improve error message on upload

### PR DESCRIPTION
**Pull request by ewdurbin**

### CODE REVIEW

File: warehouse/accounts/security_policy.py [hunk 1]
```diff
@@ -332,22 +332,29 @@ def _check_for_mfa(request, context) -> WarehouseDenied | None:
...
-        request.matched_route.name == "forklift.legacy.file_upload"
-        or request.matched_route.name.startswith("manage")
-        and request.matched_route.name != "manage.account"
-        and not any(
-            request.matched_route.name.startswith(route) for route in _exempt_routes
-        )
-        and not request.identity.has_two_factor
-    ) and (
-        # Start enforcement from 2023-08-08, but we should remove this check
-        # at the end of 2023.
```
Comment (miketheman) [inline warehouse/accounts/security_policy.py:344]: We lose this comment ass a result - intended?

**Review by miketheman — APPROVED** [review]

File: warehouse/accounts/security_policy.py (hunk 1 above)
Comment (ewdurbin) [reply warehouse/accounts/security_policy.py:344]: nope! I'm just messy.

Comment (ewdurbin) [reply warehouse/accounts/security_policy.py:344]: 8e45163457fd8c6e91ac35b2481100f175ba5981

**Comment by ewdurbin** [comment]
I'd merge it if I could :-D

**Pull request merged**